autoload/vim_git_backup/git_helper.vim

```diff
@@ -81,20 +81,17 @@ endfunction
 "     str: The found git repository root. If not root was found, return an empty string.
 "
 function! s:GetGitRoot(path)
-    " Note: This function only works on Linux!
-    let l:parts = split(a:path, '/')
-    let l:index = 0
-    let l:length = len(parts)
+    let l:current_dir = a:path
+    let l:root_dir = has('win32') ? 'C:\\' : '/'
 
-    for part in l:parts
-        let l:root = join([''] + parts[:l:length - l:index], '/')
-
-        if finddir('.git', l:root) != "" || findfile('.git', l:root) != ""
-            return l:root
+    while l:current_dir != l:root_dir
+        if filereadable(l:current_dir . '/.git') || isdirectory(l:current_dir . '/.git')
+            return l:current_dir
         endif
 
-        let index += 1
-    endfor
+        " Go up one directory level
+        let l:current_dir = fnamemodify(l:current_dir, ':h')
+    endwhile
 
     return ''
```

Comment:
Rather than relying on a fixed drive letter as a root we can make it more generic if we lean onto regex

```vim
function! s:GetGitRoot(path)
    let l:current = a:path
    let l:previous = l:current . 'extra_string_to_make_it_different_from_current'

    while l:current =~# '^\([A-Za-z]:\\\|/\)$' && l:current != l:previous_dir
        if filereadable(l:current . '/.git') || isdirectory(l:current . '/.git')
            return l:current
        endif

        " Go up one directory level
        let l:previous = l:current
        let l:current = fnamemodify(l:current, ':h')
    endwhile

    return ''
endfunction
```

Now it doesn't matter which OS calls it, the path will behave as expected.